services/server/server.go

```diff
@@ -116,81 +117,34 @@ func (s *Server) UpdateSettingsFromEnv(env []string) error {
 			return err
 		}
 
-		for _, e := range env {
-			p := strings.SplitN(e, "=", 2)
-			if len(p) != 2 {
-				s.l.Warnf("Failed to parse environment variable %q.", e)
-				continue
+		envSettings, errs, warns := validators.EnvVarValidator(env)
+		for _, w := range warns {
+			s.l.Warnln(w)
+		}
+
+		if len(errs) > 0 {
+			for _, e := range errs {
```

Comment:
It should not be possible, right? Let's add a comment for the future readers. `UpdateSettingsFromEnv` comment probably could be tweaked too.